x-pack/auditbeat/module/system/login/utmp_c.go

```diff
@@ -0,0 +1,120 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build linux
+
+// Pure Go reader for UTMP formatted files.
+// See utmp(5) and getutent(3) for the C structs and functions this is
+// replacing.
+
+package login
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+	"time"
+	"unsafe"
+)
+
+var byteOrder = getByteOrder()
+
+func getByteOrder() binary.ByteOrder {
```

Comment:
Turns out gosigar has a GetEndian() method for this.

https://github.com/elastic/beats/blob/f28cf30de74572b9323d616f16fffed177ba99c1/vendor/github.com/elastic/gosigar/sys/endian.go#L8

Reply:
Oh nice, thanks!